apps/frontend/src/components/TimeCardPage/SubmitCard.tsx

```diff
@@ -1,61 +1,214 @@
+import moment from "moment";
+import { useToast } from "@chakra-ui/react";
+import { UserContext } from "./UserContext";
+import { TimesheetStatus } from "src/schemas/TimesheetSchema";
+
+interface submitCardProps {
+  timesheetId: number;
+  associateId: string;
+  timesheetStatus: StatusType;
+  refreshTimesheetCallback: Function;
 }
 
-export default function SubmitCard({
-    setWeeklyComments,
-    setWeeklyReports,
-    weeklyComments,
-    weeklyReports
-}: SubmitCardProps) {
+export default function SubmitCard(props: submitCardProps) {
+  const toast = useToast();
+  const currUser = useContext(UserContext);
 
+  /** Whether or not the logged-in user has submitted this timesheet yet.*/
   const [submitted, setSubmitted] = useState(false);
+
+  /** The date and time (as a moment) that the logged-in user submitted/reviewed/finalized this timesheet.*/
   const [submitDate, setSubmitDate] = useState(null);
+
+  /**
+   *   The card state which corresponds to the latest status update from the timesheet. Corresponds to card color.
+   *   Note that this is *not* dependent on the logged in user. I.e. if the latest status update was that
+   *   the supervisor had submitted their timesheet review, the card state would be CardState.InReviewAdmin for
+   *   any associate, supervisor, or admin that was viewing the timesheet.
+   */
   const [state, setState] = useState(CardState.Unsubmitted);
 
+  // Run whenever there's an update to the current logged in user (i.e. re-render the correct submission status for the user type)
   useEffect(() => {
-    //TODO - API Call to determine if the table has been submitted or not.
-    //Will set submitted? here and also submitDate if it was submitted to grab the date     
-  }, [])
-
-  const submitAction = () => {
-    setSubmitted(!submitted);
-    const currentTime = new Date();
-    setSubmitDate(currentTime.toString());
-    if (state === CardState.Unsubmitted) {
-      setState(CardState.InReviewEmployer);
+    if (currUser === undefined) {
+      return;
     }
-    else {
+
+    let statusEntry: StatusEntryType = undefined;
+
+    // Determine the appropriate status entry to match up with the logged in user's role
+    switch (currUser.Type) {
+      case UserTypes.Associate:
+        statusEntry = props.timesheetStatus.HoursSubmitted;
+        break;
+
+      case UserTypes.Supervisor:
+        statusEntry = props.timesheetStatus.HoursReviewed;
+        break;
+
+      case UserTypes.Admin:
+        statusEntry = props.timesheetStatus.Finalized;
+        break;
+    }
+
+    const isSubmitted = statusEntry !== undefined;
+    setSubmitted(isSubmitted);
+
+    // Set the submitted date to when the corresponding status entry was logged for the user type
+    if (isSubmitted && statusEntry.Date !== undefined) {
+      setSubmitDate(moment.unix(statusEntry.Date));
+    }
+
+    // Determine the latest status update to set the card state
+    if (props.timesheetStatus.Finalized !== undefined) {
```

Comment:
Can we modify StatusSchema to sync with CardState? Is there a reason we don't want to do that?

Reply:
The main issue is that `StatusSchema` isn't an enum - it's an object full of `StatusEntries` for each type of submission that can occur. I agree that `CardState` can probably be renamed, but there's not a clear way that I can see `StatusSchema` translating to an enum other than what we do with CardState. 

The other option is that we could remove the state variable all together or have it be a copy of the `props.timesheetStatus` object, and then replace the color selection code with the checks we do in this `useEffect`:

```
          cardState === CardState.AdminFinalized
            ? "success"
            : cardState === CardState.InReviewAdmin ||
              cardState === CardState.InReviewSupervisor
            ? "warning"
            : "danger"
        }
```

I think it's cleaner to read with the enum, and it's helpful to have an easy variable to access the 'current' stage a ticket is in, which we otherwise don't have.